bp_common/software/py/nbf.py

```diff
@@ -191,10 +193,11 @@ def dump(self):
   parser = argparse.ArgumentParser()
   parser.add_argument('ncpus', type=int, help='number of BlackParrot cores')
   parser.add_argument('ucode_file', metavar='ucode.mem', help='CCE ucode file')
```

Comment:
Unrelated, but should be little effort.  Can you make the ucode_file optional here as well?

Reply:
But why? Is there a way to load the ucode right now except for the nbf loader?

Reply:
Well actually now that's also true for the mem argument haha.